llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -312,20 +314,13 @@ struct Frame {
   // hashing from llvm ADT since we are going to persist the hash id, the hash
   // combine algorithm in ADT uses a new randomized seed each time.
   inline FrameId hash() const {
-    auto HashCombine = [](auto Value, size_t Seed) {
-      std::hash<decltype(Value)> Hasher;
-      // The constant used below is the 64 bit representation of the fractional
-      // part of the golden ratio. Used here for the randomness in their bit
-      // pattern.
-      return Hasher(Value) + 0x9e3779b97f4a7c15 + (Seed << 6) + (Seed >> 2);
-    };
-
-    size_t Result = 0;
-    Result ^= HashCombine(Function, Result);
-    Result ^= HashCombine(LineOffset, Result);
-    Result ^= HashCombine(Column, Result);
-    Result ^= HashCombine(IsInlineFrame, Result);
-    return static_cast<FrameId>(Result);
+    llvm::HashBuilder<llvm::TruncatedBLAKE3<8>, llvm::endianness::little>
```

Comment:
What is the performance implication of using a cryptographic hash function?

Reply:
`llvm-profdata` takes 0.31% more time over five runs on a large (~600MB) raw profile.

67.782 seconds (original) v.s. 67.991 seconds (this PR)

Reply:
There were some discussions wrt `hash_combine` in the past, I do not recall what was the outcome? Looks like some alternative solution was proposed?

Reply:
I am not aware of the discussion.  Do you happen to have a pointer?

Reply:
I cannot find the references quickly (though there was a big discussion), but at least we do have `stable_hash_combine` these days, see https://github.com/llvm/llvm-project/pull/105849

Reply:
Thanks!  As I noted in https://github.com/llvm/llvm-project/pull/109501#issuecomment-2370345195, I'm planning to use BLAKE as a short-term solution.  I'm going to replace it with something else in the long term.